soteria-rust/lib/sptr.ml

```diff
@@ -164,4 +167,15 @@ module ArithPtr : S with type t = arithptr_t = struct
 
   let as_id { ptr; _ } = Typed.cast @@ Typed.Ptr.loc ptr
   let allocation_info { size; align; _ } = (size, align)
+
+  let iter_vars { ptr; align; size; _ } f =
+    Typed.iter_vars ptr f;
+    Typed.iter_vars align f;
+    Typed.iter_vars size f
```

Comment:
```suggestion
  let iter_vars { ptr; align; size; tag = _ } f =
    Typed.iter_vars ptr f;
    Typed.iter_vars align f;
    Typed.iter_vars size f
```

Nitpick but for these kinds of situations I tend to avoid the `_` pattern because if some fields are added, `iter_vars` will be ignoring them silently leading to all kinds of bugs.
This is as opposed to e.g. `allocation_info` above, where the `_` is ok because